robot_smach_states/src/robot_smach_states/world_model.py

```diff
@@ -237,20 +240,37 @@ def __init__(self, robot, segmented_entity_ids_designator, entity_designator, vo
         self.robot = robot
         self.seen_entities_des = segmented_entity_ids_designator
         self.entity_des = entity_designator
-        self.volume = volume
+
+        ds.check_type(volume_designator, str)
+        if isinstance(volume_designator, str):
+            self.volume_designator = ds.VariableDesignator(volume_designator)
+        elif isinstance(volume_designator, ds.Designator):
+            self.volume_designator = volume_designator
+        else:
+            raise RuntimeError("This shouldn't happen. Wrong types should have raised an exception earlier")
```

Comment:
I think this is a bit backwards. If I input a string now it will be transformed to a designator only to be turned back into a string when I need it.

Also, I believe that we do this 'it can be a thing or a designator to a thing' often enough that it warrants its own support function much like check_type.

Reply:
Agreed.